provisioning/hare_setup.py

```diff
@@ -0,0 +1,279 @@
+                exit(-1)
+
+            cluster_status = check_cluster_status()
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+            if cluster_status:
+                logging.error('Cluster status reports failure')
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    return os.system('rpm -qa | grep -q ' + rpm_name)
+
+
+def is_cluster_running() -> bool:
+    if os.system('hctl status >/dev/null') == 0:
+        return True
+    else:
+        return False
+
+
+def bootstrap_cluster():
+    return os.system('hctl bootstrap --mkfs /var/lib/hare/cluster.yaml')
+
+
+def shutdown_cluster():
+    return os.system('hctl shutdown')
+
+
+def list2dict(nodes_data_hctl: list) -> dict:
+    node_info_dict = {}
+    for node in nodes_data_hctl:
+        node_svc_info = {}
+        for service in node['svcs']:
+            node_svc_info[service['name']] = []
+            if (service['status'] == 'started'):
+                node_svc_info[service['name']].append(service['status'])
+        node_info_dict[node['name']] = node_svc_info
+
+    return node_info_dict
+
+
+def check_cluster_status():
+    cluster_desc = None
+    with open("/var/lib/hare/cluster.yaml", 'r') as stream:
+        cluster_desc = yaml.safe_load(stream)
+    cmd = ['hctl', 'status', '--json']
+    cluster_info_out, err = execute_command_and_get_output(cmd)
+    if err:
+        logging.error('Failed to get hctl status')
+        return -1
+
+    cluster_info = json.loads(cluster_info_out)
+    nodes_data_hctl = cluster_info['nodes']
+
+    node_info_dict = list2dict(nodes_data_hctl)
+    for node in cluster_desc['nodes']:
+        m0client_s3_cnt = node['m0_clients']['s3']
+        m0ds = node.get('m0_servers', [])
+        for m0d in m0ds:
+            ios_cnt = 0
+            if ('runs_confd' in m0d.keys() and
+                (node_info_dict[node['hostname']]['confd'][0] != 'started')):
+                return -1
+            if m0d['io_disks']['data']:
+                if node_info_dict[node['hostname']]['ioservice'][ios_cnt] != 'started':
+                  return -1
+                ios_cnt += 1
+        if (m0client_s3_cnt > 0 and
+            len(node_info_dict[node['hostname']]['s3server']) != m0client_s3_cnt):
+                return -1
+
+    return 0
+
+
+def main(argv=None):
```

Comment:
I guess `argv` are always None. Probably you should remove them from the signature.

Reply:
Done